cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -75,7 +75,14 @@ func NewCommand() *cobra.Command {
 			namespace, _, err := clientConfig.Namespace()
 			errors.CheckError(err)
 
-			resyncDuration := time.Duration(appResyncPeriod) * time.Second
+			var resyncDuration time.Duration
+			if appResyncPeriod == 0 {
+				// Re-sync should be disabled if period is 0. Set duration to a very long duration
+				resyncDuration = time.Hour * 24 * 365
```

Comment:
I haven't looked at the code, but how does the controller figure out _when_ to reconcile again? I am assuming it takes the timestamp from application's `.status.reconciledAt` and checks whether `now - reconciledAt > resyncDuration`, right?

I am wondering whether it's sufficient to set the `resyncDuration` to a year. I am imagining the following (admittedly, very far edge) scenario:

* user has 1000 apps that get synced once
* user disables resync
* no further syncs happen (e.g. applications were synced once, never changed)
* after a year, appSyncPeriod expires and apps will get resynced

That might end up in some very nasty surprise. A year is not that long of a period. Maybe we should change it to 100 years? :)

Reply:
I agree. 100 years sounds pretty safe :) updated

Reply:
Then the surprise will be for future generations still using argocd :)